finality-aleph/src/aggregation/aggregator.rs

```diff
@@ -131,23 +202,20 @@ where
     pub(crate) async fn next_multisigned_hash(&mut self) -> Option<(H, MK::PartialMultisignature)> {
         loop {
             trace!(target: "aleph-aggregator", "Entering next_multisigned_hash loop.");
-            match self.hash_queue.front() {
-                Some(hash) => {
-                    if let Some(multisignature) = self.signatures.remove(hash) {
-                        let hash = self
-                            .hash_queue
-                            .pop_front()
-                            .expect("VecDeque::front() returned Some(_), qed.");
-                        return Some((hash, multisignature));
-                    }
+            match self.aggregator.try_pop_hash() {
+                Ok(res) => {
+                    return Some(res);
                 }
-                None => {
-                    if self.last_hash_placed {
-                        debug!(target: "aleph-aggregator", "Terminating next_multisigned_hash because the last hash has been signed.");
-                        return None;
-                    }
+                Err(AggregatorError::LastHashPlaced) => {
+                    debug!(
+                        target: "aleph-aggregator",
+                        "Terminating next_multisigned_hash because the last hash has been signed."
+                    );
+                    return None;
                 }
+                Err(_) => {}
```

Comment:
This looks mysterious, perhaps spell out the other kind of error you expect here (`NoHashFound`) and ignore it, while logging any unexpected error as problematic (although you can likely ignore it in the logic, it'll never happen anyway).